### PR TITLE
Implemented RuboCop, Ruby-style linter.

### PR DESCRIPTION
**Pull request by dblock**
How does everyone feel about adding RuboCop?

### CODE REVIEW

**Comment by dblock** [comment]
cc: @digitalplaywright @johnnyshields

**Comment by digitalplaywright** [comment]
I think that is a great idea. I've used RuboCop in other projects with great success.

**Comment by dblock** [comment]
I'll merge once the build is green.

File: lib/mongoid/slug/criteria.rb [hunk 1]
```diff
@@ -75,20 +75,23 @@ def objectid_slug_strategy id
...
-
       def for_slugs(slugs)
-        #_translations
-        localized = (@klass.fields['_slugs'].options[:localize] rescue false)
+        # _translations
+        localized = (begin
+                       @klass.fields['_slugs'].options[:localize]
+                     rescue
+                       false
+                     end)
```
Comment (johnnyshields) [inline lib/mongoid/slug/criteria.rb]: Previous was was more legible IMHO.

Comment (dblock) [inline lib/mongoid/slug/criteria.rb]: I think this is a great example why RuboCop is great. You just run `rubocop -a` and let Ruby style guide tell you what's more legible. This way we don't have to have to argue about it :)

Personally half the time I don't see that there's a `rescue` there and wonder how come `localized` is `false` when debugging.

**Pull request merged**